cmd/notary/integration_test.go

```diff
@@ -81,6 +81,43 @@ func setupServer() *httptest.Server {
 	return httptest.NewServer(setupServerHandler(storage.NewMemStorage()))
 }
 
+// Initializes a repo with existing key
+func TestInitWithMultipleRootKeys(t *testing.T) {
+	// -- setup --
+	setUp(t)
+
+	tempDir := tempDirWithConfig(t, "{}")
+	defer os.RemoveAll(tempDir)
+
+	server := setupServer()
+	defer server.Close()
+
+	tempFile, err := ioutil.TempFile("", "targetfile")
+	require.NoError(t, err)
+	tempFile.Close()
+	defer os.Remove(tempFile.Name())
+
+	// -- tests --
+
+	// generate root key produces a single root key and no other keys
+	_, err = runCommand(t, tempDir, "key", "generate", data.ECDSAKey)
+	require.NoError(t, err)
+	rootIDs, _ := assertNumKeys(t, tempDir, 1, 0, true)
+	var rootFiles []string
+	for _, rID := range rootIDs {
+		rootFiles = append(rootFiles, filepath.Join(
+			tempDir, "private", "root_keys", rID+".key"))
+	}
+
+	// init repo
+	_, err = runCommand(t, tempDir, "-s", server.URL, "init", "gun", "--rootkey", rootFiles[0])
+	require.NoError(t, err)
+
+	// publish repo
+	_, err = runCommand(t, tempDir, "-s", server.URL, "publish", "gun")
+	require.NoError(t, err)
```

Comment:
@ecordell  You are definitely right in that the utility functions you use for loading the key should all have be tested already.  I agree with you that it's pointless to re-test this functionality here; I  mainly wanted to test that particular loading checks (like ensuring that we only allow importing encrypted keys - we did not have to include this check, after all) are called and that their errors are propagated (for instance that we don't just fall back on generating a random root key if importing a key fails), and wanted some way to trigger said failures.  

For instance, for case 3, it's to test that the root key is required to be encrypted - you could for instance completely skip that check and allow importing an unencrypted root key (which may or may not be a valid choice).

Also, for case 4, cryptoservice's `AddKey` will definitely fail if the roles are mismatched, but I'm also wondering if we should do a separate check, in case you are importing a targets key from a different location (for instance, you might have a separate `~/.docker/trust` and `~/.notary` directory with different keys, and the `AddKey` check will only fail if you import a targets key from `~/.notary` into a repo in `~/.notary`, but won't fail if you import a targets key that exists only in `~/.docker/trust` into `~/.notary`).  But maybe this case doesn't matter as much.